diff --git a/node_modules/connect/index.js b/node_modules/connect/index.js
index 5f2e578..3c33279 100644
--- a/node_modules/connect/index.js
+++ b/node_modules/connect/index.js
@@ -83,8 +83,11 @@ app.use = function use(route, fn) {
   }

   debug('use %s %s', route || '/', fn.name || 'anonymous')
-
-  this.stack.push(new Layer(route, fn))
+  
+  // Add a check to ensure the fn argument has a handle function
+  if (fn && (typeof fn === 'function' || typeof fn.handle === 'function')) {
+    this.stack.push(new Layer(route, fn))
+  }

   return this
 }
